Первый коммит по поиску релевантного файла

r-i-besedin opened the pull request:
Файлы для индексации вводятся без ключа аргументов, например:
        ./main ./search ex1.txt ex2.txt ex3.txt
Поисковая фраза - строка с ключевыми словами, разделенными пробелами
Программа выведет файлы в порядке наилучшего соответствия (слова из поисковой фразы встретились в файле наибольшее количество раз).

r-i-besedin committed:
Первый коммит по поиску релевантного файла

Файлы для индексации вводятся без ключа аргументов, например:
        ./main ./search ex1.txt ex2.txt ex3.txt
Поисковая фраза - строка с ключевыми словами, разделенными пробелами
Программа выведет файлы в порядке наилучшего соответствия (слова из поисковой фразы встретились в файле наибольшее количество раз).

tariel-x commented on main.go:
```diff
@@ -0,0 +1,81 @@
+import (
+	"bufio"
+	"fmt"
+	"./invertedindex"
```
`"github.com/polis-mail-ru-golang-1/t2-invert-index-search-RGrouse/invertedindex"`

https://golang.org/doc/code.html#ImportPaths

Как быть с fork можно почитать здесь: http://code.openark.org/blog/development/forking-golang-repositories-on-github-and-managing-the-import-path

tariel-x commented on main.go:
```diff
@@ -0,0 +1,81 @@
+	m := make(map[string]int)
+
+	for scanner.Scan() {
+		m[scanner.Text()]++
```
Подсчёт количества встретившихся слов лучше вынести в пакет с индексом, т.к. это больше относится к логике построения его. Разбиение по словам можно оставить здесь (на случай использования лемматизатора), можно тоже перенести в пакет.

tariel-x commented on invertedindex/invertedindex.go:
```diff
@@ -0,0 +1,49 @@
+}
+
+//карта вида: слово - массив из записей {источник, сколько раз слово употреблено в источнике}
+var globalindexmap map[string]*[]entry
```
Зачем здесь ссылка `*[]entry`?

r-i-besedin replied:
Перенес всё, создал новый pool request, ссылка действительно незачем, убрал ее

tariel-x commented:
Настройте, пожалуйста, http://jen20.com/2015/09/14/gofmt-or-goimports-in-intellij.html

r-i-besedin committed:
Подсчет количества слов вынесен в пакет с индексом

Теперь в главном пакете формируется слайс слов из файла, а подсчитываются они уже в пакете с индексом
